fabric-shim/lib/contract-spi/chaincodefromcontract.js

```diff
@@ -255,6 +255,20 @@ class ChaincodeFromContract {
                     transaction.tags = ['submitTx'];
                 }
 
+                // add 'argN' parameters, skipping the first (stub) parameter
+                if (propValue.length > 1) {
+                    transaction.parameters = [];
+                    for (let param = 1; param < propValue.length; param++) {
+                        transaction.parameters.push({
+                            name: `arg${param - 1}`,
+                            description: `Argument ${param - 1}`,
+                            schema: {
+                                type: 'string'
```

Comment:
Should this be string? There is no guarentee that the TX function will want a string, if they have a custom serializer they may convert it to another type, could we leave this blank?

Reply:
I see your point; we'd need to change master & 2.0 as well.

Reply:
though thinking more about it @awjh-ibm, that value is only passed to the serializer as this is what is expected; it'll have to work everything out itself in JavaScript anyway.  So I think it would be tempted to leave it pending user feedback :-)

Reply:
There is the risk that the serializer won't return a string, but since its in 2 we'll stick with the same functionality and can update if required. People can always manually update their metadata if needs be as I think they will be the edge cases